addons/udes_stock_picking_batch/models/stock_picking_batch.py

```diff
@@ -885,23 +842,21 @@ def assign_batch(self, picking_type_id, selection_criteria=None):
         """
         Determine all the batches in state 'ready' with pickings
         of the specified picking types then return the one determined
-        by the selection criteria method (that should be overriden
+        by the selection criteria method (that should be overridden
         by the relevant customer modules).
 
         Note that the transition from state 'ready' to 'in_progress'
         is handled by computation of state function.
         """
-        batch = self.env["stock.picking.batch"]
-        
-        batches = self.search([("state", "=", "ready")]).filtered(
-            lambda b: all([pt.id == picking_type_id for pt in b.picking_type_ids])
-        )
+        if not type(picking_type_id) == int:
+            raise ValidationError("picking_type_id should be an integer.")
+        batches = self.search([("state", "=", "ready"), ("picking_type_id", "=", picking_type_id)])
 
         if batches:
             batch = self._select_batch_to_assign(batches)
             batch.user_id = self.env.user
 
-        return batch
+        return batches
```

Comment:
Question: DO we need to return batches (can be more than one), or the fix should be to initialise batch = False before if batches: and return batch?

Reply:
This is actually a mistake of mine, I somehow thought we where reassigning batches but clearly we are not. I will fix it to return batch. Good spot! 👍🏻